src/main/kotlin/org/rust/cargo/project/model/CargoProjectService.kt

```diff
@@ -92,9 +95,20 @@ interface CargoProject : UserDataHolderEx {
     val stdlibStatus: UpdateStatus
     val rustcInfoStatus: UpdateStatus
 
-    val mergedStatus: UpdateStatus get() = workspaceStatus
-        .merge(stdlibStatus)
-        .merge(rustcInfoStatus)
+    val mergedStatus: UpdateStatus
+        get() = workspaceStatus
+            .merge(stdlibStatus)
+            .merge(rustcInfoStatus)
+
+    val userDisabledFeatures: UserDisabledFeatures
+
+    /**
+     * Modifies [userDisabledFeatures] that eventually affects [CargoWorkspace.Package.featureState].
+     * Note that [CargoProject] is immutable object, so this instance is not mutated by [modifyFeatures]
+     * invocation. Instead, new [CargoProject] instance is created and replaces this instance in
+     * [CargoProjectsService.allProjects].
+     */
+    fun modifyFeatures(features: Set<PackageFeature>, newState: FeatureState)
```

Comment:
Do we really need it?
Maybe it's better just use `org.rust.cargo.project.model.impl.CargoProjectsServiceImpl#modifyFeatures` directly? (of course, it requires moving `modifyFeatures` into `CargoProjectsService` intefrace)

Reply:
I don't like it because it creates the impression that it's possible to modify cargo project. And you even had to write the corresponding doc comment because of it. At the same time, it just delegates all logic to `CargoProjectsServiceImpl` so I'd prefer not to introduce potentially confusing API